src/common/tagging/external/tag_group.go

```diff
@@ -0,0 +1,209 @@
+	newTags, existingTags := block.GetNewTags(), block.GetExistingTags()
+	newTags = append(newTags, existingTags...)
+	satisfyFilters := true
+	for filterKey, filterValue := range t.filters {
+		if filterKey == "tags" {
+			for filterTagKey, filterTagValue := range filterValue.(map[interface{}]interface{}) {
+				strFilterValue := filterTagValue
+				if val, ok := filterTagValue.(int); ok {
+					strFilterValue = strconv.Itoa(val)
+				}
+				foundFilterTag := false
+				for _, blockTag := range newTags {
+					if blockTag.GetKey() == filterTagKey && blockTag.GetValue() == strFilterValue {
+						foundFilterTag = true
+						break
+					}
+				}
+				satisfyFilters = satisfyFilters && foundFilterTag
+			}
+		}
+		if filterKey == "directory" {
+			if tagFilterDir != filterValue {
+				satisfyFilters = false
+				break
+			}
+		}
+	}
+	return satisfyFilters
+}
+
+func (t *TagGroup) InitExternalTagGroups(configFilePath string) {
+	t.configFilePath = configFilePath
+	t.tagGroupsByName = make(map[string][]Tag)
+	t.InitExternalTagGroup()
+
+}
+
+func (t *TagGroup) InitTagGroup(dir string, skippedTags []string) {
+	t.SkippedTags = skippedTags
+	t.Dir = dir
+}
+
+func (t *TagGroup) InitExternalTagGroup() {
+	configMap := make(map[interface{}]interface{})
+	confBytes, err := ioutil.ReadFile(t.configFilePath)
+	if err != nil {
+		logger.Error(err.Error())
+	}
+	errYaml := yaml.Unmarshal(confBytes, &configMap)
+	if errYaml != nil {
+		logger.Error(errYaml.Error())
+	}
+	t.config = configMap
+	t.extractExternalTags()
+}
+
+func (t *TagGroup) extractExternalTags() {
+	tagGroups := t.config["tag_group"]
+	switch tgs := tagGroups.(type) {
+	case map[interface{}]interface{}:
+		groupName := tgs["name"].(string)
+		t.tagGroupsByName[groupName] = t.ExtractExternalGroupsTags(tgs["tags"].([]interface{}))
+	case []interface{}:
+		for _, tagGroup := range tgs {
+			tagGroupMap := tagGroup.(map[interface{}]interface{})
+			groupName := tagGroupMap["name"].(string)
+			t.tagGroupsByName[groupName] = t.ExtractExternalGroupsTags(tagGroupMap["tags"].([]interface{}))
+		}
+	}
+}
+
+func (t *TagGroup) GetDefaultTags() []tags.ITag {
+	return []tags.ITag{}
+}
+
+func (t *TagGroup) CreateTagsForBlock(block structure.IBlock) error {
+	newTags, existingTags := block.GetNewTags(), block.GetExistingTags()
+	var filteredNewTags = make([]tags.ITag, len(newTags))
+	copy(filteredNewTags, newTags)
+	for _, groupTags := range t.tagGroupsByName {
+		for _, groupTag := range groupTags {
+			tagValue, err := t.calculateTagValue(block, groupTag)
+			if err != nil {
+				logger.Error(err.Error())
+			}
+			if tagValue == nil {
+				for i, newTag := range newTags {
+					if newTag.GetKey() == groupTag.GetKey() {
+						filteredNewTags = append(filteredNewTags[:i], filteredNewTags[i+1:]...)
+					}
+				}
+			} else {
+				filteredNewTags = append(filteredNewTags, tagValue)
+			}
+		}
+	}
+	filteredNewTags = append(filteredNewTags, existingTags...)
+	t.SetTags(filteredNewTags)
```

Comment:
@tronxd Why is this necessary? This should be part of the setup, to support the skip functionality